### ISSUE TITLE
[Bug] angle_axis_to_rotation_matrix can not be backward

### ISSUE THREAD

**Issue by tuanhui-li** [kornia/kornia#611]
## ❓ Questions and Help
I want to use the angle_axis_to_rotation_matrix as a function in the pytorch pipline, the demo code is as follows:
```
import torch
import torchgeometry as tgm

x = torch.zeros(1, 3, requires_grad=True)  # Nx3
input = tgm.deg2rad(x)
output = tgm.angle_axis_to_rotation_matrix(input)  # Nx4x4
output.mean().backward()
print(x.grad)
```
then the result is as follows:
```
tensor([[nan, nan, nan]])
```

**Comment by edgarriba** [comment]
@tuanhui-li from your example I can see that your are using `torchgeometry` which is a very old version of what kornia is right now. 

Could you try with recent versions ?

**Comment by tuanhui-li** [comment]
@edgarriba I tried kornia , and I got the same result

**Comment by edgarriba** [comment]
@tuanhui-li rotations API needs to be revisited since we have been seeing a lot of issues recently.

**Comment by cbames** [comment]
In  **_compute_rotation_matrix(angle_axis, theta2, eps=1e-6):**    
`
    k_one = 1.0  `
`    theta = torch.sqrt(theta2)  `
`    wxyz = angle_axis / (theta + eps)  `

if theta2 is zero, then on a backwards call, the gradient will become infinite. A simple fix is to add the already previously defined eps to theta2 instead of to theta. Although that could result in theta being very small compared to eps, so I've included theta2+eps and theta+eps.

**Comment by edgarriba** [comment]
@cbames thanks ! could you send a small PR fixing this ? can we have a minimal test to verifying this ?

**Comment by cbames** [comment]
Sure thing, I'll toss it on the todo list.

**Comment by Zju-George** [comment]
@cbames 
https://github.com/kornia/kornia/blob/009a3929a894757363218558d09ed7bfc41c9c3d/kornia/geometry/conversions.py#L242
change to 'theta = torch.sqrt(theta2 + eps)'?

**Comment by edgarriba** [comment]
@Zju-George worth checking/suggesting in https://github.com/kornia/kornia/pull/926
